app/hooks/MenuHooks.ts

```diff
@@ -0,0 +1,45 @@
+import { useState, useRef, useEffect } from "react";
+
+export function useMenuToggle() {
+  const [activeSubMenu, setActiveSubMenu] = useState<string | null>(null);
```

Comment:
Why isn't this property a boolean? I'm a little confused about this design. The fact that this is a string/identifier suggests there is code that is managing the current active menu across multiple dropdowns, but I don't see that code here. Or that this component has to be aware of other dropdowns in the same render scope (ie, on the page). I would expect this property to be something like `isActive`. Maybe I'm missing something 😅